populator-machinery/controller.go

```diff
@@ -291,6 +297,8 @@ func RunControllerWithConfig(vpcfg VolumePopulatorConfig) {
 	c.metrics.startListener(vpcfg.HttpEndpoint, vpcfg.MetricsPath)
 	defer c.metrics.stopListener()
 
+	c.prometheusMetrics.startListener(vpcfg.PrometheusHttpEndpoint, vpcfg.PrometheusMetricsPath)
```

Comment:
I think this should be optional? Not everyone using prometheus?

Reply:
The listener is already optional, is it will no-op if either the PrometheusHttpEndpoint or the PrometheusMetricsPath strings are empty. https://github.com/kubernetes-csi/lib-volume-populator/pull/172/files#diff-08529a0c4885fa455eabd360c95bd6fdbaa99b523657127a4903ede7956f2ad0R242.

Is your suggestion to make it more explicit that it is optional? (perhaps by taking the above validation outside of the function and placing it above the line you highlighted).

Reply:
If the PrometheusHttpEndpoint/PrometheusMetricsPath is empty string, I assume prometheus will not get initialized? will c.prometheusMetrics.recordError return error in this case?